cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/helpers.js

```diff
@@ -0,0 +1,415 @@
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowToggleColumnButtonHasIntendedBehavior() {
+  getExpandedDocRowFieldName(0, 0).then(($expandedDocumentRowFieldText) => {
+    const fieldText = $expandedDocumentRowFieldText.text();
+    getExpandedDocRow(0, 0).within(() => {
+      cy.getElementByTestId('docTableHeader-' + fieldText).should('not.exist');
+      cy.getElementByTestId('toggleColumnButton').click();
+    });
+    cy.getElementByTestId('fieldList-selected').within(() => {
+      cy.getElementByTestId('field-' + fieldText).should('exist');
+    });
+    cy.getElementByTestId('docTableHeader-' + fieldText).should('exist');
+    cy.getElementByTestId('fieldToggle-' + fieldText).click();
+    cy.getElementByTestId('fieldList-selected').within(() => {
+      cy.getElementByTestId('field-' + fieldText).should('not.exist');
+    });
+    cy.getElementByTestId('docTableHeader-' + fieldText).should('not.exist');
+  });
+}
+
+/**
+ * Check the first expanded Doc Table Field's first row's Exists Filter button filters the correct Field.
+ * @param {number} docTableRowNumber Integer starts from 0 for the first row
+ * @param {number} expandedDocumentRowNumber Integer starts from 0 for the first row
+ * @param {string} expectedQueryHitsWithoutFilter expected number of hits in string after the filter is removed Note you should add commas when necessary e.g. 9,999
+ * @param {string} expectedQueryHitsAfterFilterApplied expected number of hits in string after the filter is applied. Note you should add commas when necessary e.g. 9,999
+ */
+export function verifyDocTableFirstExpandedFieldFirstRowExistsFilterButtonFiltersCorrectField(
+  docTableRowNumber,
+  expandedDocumentRowNumber,
+  expectedQueryHitsWithoutFilter,
+  expectedQueryHitsAfterFilterApplied
+) {
+  getExpandedDocRowFieldName(docTableRowNumber, expandedDocumentRowNumber).then(
+    ($expandedDocumentRowField) => {
+      const filterFieldText = $expandedDocumentRowField.text();
+      getExpandedDocRow(docTableRowNumber, expandedDocumentRowNumber).within(() => {
+        cy.getElementByTestId('addExistsFilterButton').click();
+      });
+      // Verify full pill text
+      // globalFilterLabelValue gives the inner element, but we may want all the text in the filter pill
+      cy.getElementByTestId('globalFilterLabelValue', {
+        timeout: 10000,
+      })
+        .parent()
+        .should('have.text', filterFieldText + ': ' + 'exists');
+      cy.getElementByTestId('discoverQueryHits').should(
+        'have.text',
+        expectedQueryHitsAfterFilterApplied
+      );
+    }
+  );
+  cy.getElementByTestId('globalFilterBar').find('[aria-label="Delete"]').click();
+  cy.getElementByTestId('discoverQueryHits').should('have.text', expectedQueryHitsWithoutFilter);
+}
+
+/**
+ * Sends a new query via the query multiline editor.
+ * @param del true/false. true: Deletes character to the right of the cursor; false: Deletes character to the left of the cursor
```

Comment:
- I think this comment is outdated. This function only has `query` as argument, which is not listed as a param here
- Just FYI, i update the `setSingleLineQueryEditor` to be both used for both singleLine and multiLine here: https://github.com/opensearch-project/OpenSearch-Dashboards/pull/9112/files#diff-c71a49edcd56336affaeb1dfd1e47f29965ee58b7b9368943ea1468bb9fb307aR6 . I believe you should be able to leverage that if my PR gets merged